crates/bevy_math/src/common_traits.rs

```diff
@@ -410,3 +489,48 @@ impl_stable_interpolate_tuple!(
     (T9, 9),
     (T10, 10)
 );
+
+/// A type that has tangents.
+pub trait HasTangent {
+    /// The tangent type.
+    type Tangent: VectorSpace;
```

Comment:
So all Tangents need to be a vector space?
Hm... I suppose differential geometry does agree with that.